js/modules/k6/crypto/x509/x509.go

```diff
@@ -86,47 +131,42 @@ type PublicKey struct {
 	Key       interface{}
 }
 
-// New constructs the X509 interface
-func New() *X509 {
-	return &X509{}
-}
-
 // Parse produces an entire X.509 certificate
-func (X509) Parse(ctx context.Context, encoded []byte) Certificate {
+func (mi X509) Parse(encoded []byte) Certificate {
 	parsed, err := parseCertificate(encoded)
 	if err != nil {
-		throw(ctx, err)
+		common.Throw(mi.vu.Runtime(), err)
 	}
 	certificate, err := makeCertificate(parsed)
 	if err != nil {
-		throw(ctx, err)
+		common.Throw(mi.vu.Runtime(), err)
 	}
 	return certificate
 }
 
-// GetAltNames extracts alt names
-func (X509) GetAltNames(ctx context.Context, encoded []byte) []string {
+// AltNames extracts alt names
+func (mi X509) AltNames(encoded []byte) []string {
 	parsed, err := parseCertificate(encoded)
 	if err != nil {
-		throw(ctx, err)
+		common.Throw(mi.vu.Runtime(), err)
 	}
 	return altNames(parsed)
 }
 
-// GetIssuer extracts certificate issuer
-func (X509) GetIssuer(ctx context.Context, encoded []byte) Issuer {
+// Issuer extracts certificate issuer
+func (mi X509) Issuer(encoded []byte) Issuer {
 	parsed, err := parseCertificate(encoded)
 	if err != nil {
-		throw(ctx, err)
+		common.Throw(mi.vu.Runtime(), err)
```

Comment:
I kind of prefer to return `error` instead, it does the same thing, but it still IMO looks better, so we might think about it.